scripts/add_json.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+IMAGE_HASH=$(sha256sum "$BIN_DIR/$IMAGE_NAME" | awk '{printf($1)}')
+JSON_FILE="$TOPDIR/json/$DEVICE_ID.json"
+SEP=","
+
+mkdir -p "$TOPDIR/json/"
+
+if [ ! -f "$JSON_FILE" ]; then
+    SEP=""
+    # create new json file file
+    cat <<- EOF > $JSON_FILE
+{
+    "id": "$DEVICE_ID",
+    "title": "$DEVICE_TITLE",
+    "vendor": "$DEVICE_VENDOR",
+    "model": "$DEVICE_MODEL",
+    "variant": "$DEVICE_VARIANT",
+    "supported_devices": [ $(echo $SUPPORTED_DEVICES | sed -e 's/^/"/' -e 's/$/"/' -e 's/ /", "/') ],
```

Comment:
From my initial feeling, I would have expected supported_devices to be absent if empty. However, I think that's a matter of taste.
Speaking of taste, I would also have used hyphens here and for the other variables: supported-devices, image-size, etc., as long as they are lower-case ...

Reply:
You need to add the global parameter for the last sed pattern:
` -e 's/ /", "/g'`
Otherwise, this will only substitute the first space if three SUPPORTED_DEVICES are present.

Reply:
And instead of the first two patterns, you can just enclose everything in quotes, as
`echo "" | sed -e 's/^/"/' -e 's/$/"/'`
will yield `""` anyway.
But this behaviour is wrong anyway, as it will put `""` into the array when SUPPORTED_DEVICES is empty.

So you would want something like:
`echo $(echo "$SUPPORTED_DEVICES" | sed -E -e 's/^(.+)$/"\1"/' -e 's/ /", "/g')`

Reply:
Thanks, updated!